server/src/main/java/org/apache/druid/discovery/DruidLeaderClient.java

```diff
@@ -256,15 +257,21 @@ public String findCurrentLeader()
     );
   }
 
-  private String getCurrentKnownLeader(final boolean cached) throws IOException
```

Comment:
I am a bit hesitant of changing the type of exception here. Since callers might be catching this right now. Maybe keep it IOException for now and just change the error message.

Reply:
Yes, that makes sense. Let me just update the exception message.